protocol/x/clob/e2e/liquidation_deleveraging_test.go

```diff
@@ -20,6 +20,616 @@ import (
+			subaccounts: []satypes.Subaccount{
+				constants.Carl_Num0_1BTC_Short_100000USD,
+				constants.Dave_Num0_1BTC_Long_49501USD_Short,
+			},
+
+			placedMatchableOrders: []clobtypes.MatchableOrder{
+				&constants.Order_Carl_Num0_Id2_Clob0_Buy1BTC_Price50500_GTB10, // Order at $50,500
+			},
+			liquidatableSubaccountIds: []satypes.SubaccountId{constants.Dave_Num0},
+			liquidationConfig: clobtypes.LiquidationsConfig{
+				MaxLiquidationFeePpm: 5_000,
+				FillablePriceConfig:  constants.FillablePriceConfig_Default,
+				PositionBlockLimits:  constants.PositionBlockLimits_Default,
+				SubaccountBlockLimits: clobtypes.SubaccountBlockLimits{
+					// Subaccount may only liquidate $5,000 per block.
+					MaxNotionalLiquidated:    5_000_000_000,
+					MaxQuantumsInsuranceLost: 100_000_000_000_000,
+				},
+			},
+
+			liquidityTiers: constants.LiquidityTiers,
+			perpetuals: []perptypes.Perpetual{
+				constants.BtcUsd_20PercentInitial_10PercentMaintenance,
+			},
+			clobPairs: []clobtypes.ClobPair{constants.ClobPair_Btc},
+
+			expectedSubaccounts: []satypes.Subaccount{
+				{
+					Id: &constants.Carl_Num0,
+					AssetPositions: []*satypes.AssetPosition{
+						{
+							AssetId:  0,
+							Quantums: dtypes.NewInt(100_000_000_000 - 5_050_000_000),
+						},
+					},
+					PerpetualPositions: []*satypes.PerpetualPosition{
+						{
+							PerpetualId:  0,
+							Quantums:     dtypes.NewInt(-90_000_000), // -0.9 BTC
+							FundingIndex: dtypes.NewInt(0),
+						},
+					},
+				},
+				{
+					Id: &constants.Dave_Num0,
+					AssetPositions: []*satypes.AssetPosition{
+						{
+							AssetId:  0,
+							Quantums: dtypes.NewInt(-49_501_000_000 + 5_050_000_000 - 25_250_000),
+						},
+					},
+					PerpetualPositions: []*satypes.PerpetualPosition{
+						{
+							PerpetualId:  0,
+							Quantums:     dtypes.NewInt(90_000_000), // 0.9 BTC
+							FundingIndex: dtypes.NewInt(0),
+						},
+					},
+				},
+			},
+		},
+		`Liquidating short respects subaccount block limit - MaxQuantumsInsuranceLost`: {
+			subaccounts: []satypes.Subaccount{
+				// Carl_Num0 is irrelevant to the test, but is used to seed the insurance fund.
+				constants.Carl_Num0_1BTC_Short_50499USD,
+				constants.Carl_Num1_1BTC_Short_50499USD,
+				constants.Dave_Num0_1BTC_Long_50000USD,
+			},
+
+			placedMatchableOrders: []clobtypes.MatchableOrder{
+				// This order is irrelevant to the test, but is used to seed the insurance fund.
+				&constants.Order_Dave_Num0_Id2_Clob0_Sell1BTC_Price49500_GTB10, // Order at $49,500
+
+				// Bankruptcy price is $50,499, and closing at $50,500 would require $1 from the insurance fund.
+				// First order would transfer $0.1 from the insurance fund and would succeed.
+				// Second order would require $0.9 from the insurance fund and would fail.
+				&constants.Order_Dave_Num0_Id1_Clob0_Sell01BTC_Price50500_GTB10, // Order at $50,500
+				&constants.Order_Dave_Num0_Id0_Clob0_Sell1BTC_Price50500_GTB10,  // Order at $50,500
```

Comment:
interesting, so instead of partially matching the second order such that we do not exceed MaxQuantumsInsuranceLost, we instead don't match at all?

Reply:
yup - partially matching will require a bunch of new logic in the order matching loop and probably not worth the engineering effort. but let's keep this in mind as this may happen 

I don't have numbers to back me up but my intuition tells me that orders on the book are going to be fairly small relative to our current `MaxInsuranceFundLost` value so this shouldn't be an issue for us.

Reply:
true, thanks for clarifying